client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -1,43 +1,33 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	isEmpty = require( 'lodash/isEmpty' );
+var React = require( 'react' );
 
 /**
  * Internal dependencies
  */
 var DomainSuggestion = require( 'components/domains/domain-suggestion' ),
 	cartItems = require( 'lib/cart-values/cart-items' ),
-	DomainSuggestionFlag = require( 'components/domains/domain-suggestion-flag' );
+	Gridicon = require( 'components/gridicon' ),
+	DomainSuggestionFlag = require( 'components/domains/domain-suggestion-flag' ),
+	DomainSuggestionMixin = require( 'components/domains/mixins/domain-suggestion-mixin' );
 
-var DomainRegistrationSuggestion = React.createClass( {
+const DomainRegistrationSuggestion = React.createClass( {
 	propTypes: {
-		cart: React.PropTypes.object,
-		suggestion: React.PropTypes.object,
-		onButtonClick: React.PropTypes.func,
-		withPlansOnly: React.PropTypes.bool
+		cart: React.PropTypes.object.isRequired,
+		suggestion: React.PropTypes.object.isRequired,
+		onButtonClick: React.PropTypes.func.isRequired,
+		withPlansOnly: React.PropTypes.bool.isRequired,
+		selectedSite: React.PropTypes.object.isRequired
 	},
+	mixins: [ DomainSuggestionMixin ],
 
-	buttonLabel: function( isAdded ) {
-		if ( this.props.buttonLabel ) {
-			return this.props.buttonLabel;
-		}
-
-		if ( isAdded ) {
-			return null;
-		}
-
-		return this.translate( 'Add', {
-			context: 'Add a domain registration to the shopping cart'
-		} );
-	},
-
-	render: function() {
-		var suggestion = this.props.suggestion ? this.props.suggestion : {},
-			domainName = suggestion.domain_name ? suggestion.domain_name : this.translate( 'Loading\u2026' ),
+	render() {
+		var suggestion = this.props.suggestion || {},
+			domainName = suggestion.domain_name || this.translate( 'Loading\u2026' ),
```

Comment:
The `domainName` variable here is confusing - it's only used for presentation purposes in one place in the methods, while `suggestion.domain_name` should be (and is) used for logic, etc. I see that how it's done in the old code, but maybe just to be one the safe side (so that no one uses `domainName` for calling, say, `hasDomainInCart`), we can get rid of this variable?